### PR TITLE
Deprecate models and NLU pipelines

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Deprecated `SklearnIntentClassifier`, `KerasPolicy`
- Deprecated all pipeline templates
- Update NLU tests (reduced number of epochs for training)

(documentation will be updated in a separate PR)

related to https://github.com/RasaHQ/research/issues/62

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/policies/keras_policy.py [hunk 1]
```diff
@@ -70,6 +71,13 @@ def __init__(
 
         self.current_epoch = current_epoch
 
+        raise_warning(
+            "'KerasPolicy' is deprecated and will be removed in version "
+            "2.0. Use 'TEDPolicy' instead.",
+            category=FutureWarning,
+            docs=f"{DOCS_BASE_URL}/core/policies/",
```
Comment (dakshvar22) [inline rasa/core/policies/keras_policy.py]: Import this constant as well
```suggestion
            docs=f"{DOCS_URL_POLICIES}",
```

File: tests/nlu/conftest.py [hunk 2]
```diff
@@ -36,26 +28,85 @@ def spacy_nlp_component(component_builder, default_config):
...
 @pytest.fixture(scope="session")
-def default_config():
-    return config.load(CONFIG_DEFAULTS_PATH)
+def config_path() -> Text:
+    return write_file_config(
+        {
+            "language": "en",
+            "pipeline": [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "CRFEntityExtractor", EPOCHS: 2},
```
Comment (dakshvar22) [inline tests/nlu/conftest.py]: Can we fix random seeds as well in all of these, since we train and test for prediction, the corresponding tests should be deterministic.

File: tests/nlu/conftest.py [hunk 3]
```diff
@@ -36,26 +28,85 @@ def spacy_nlp_component(component_builder, default_config):
...
+
+@pytest.fixture()
+def pretrained_embeddings_convert_config() -> RasaNLUModelConfig:
+    return RasaNLUModelConfig(
+        {
+            "language": "en",
+            "pipeline": [
+                {"name": "ConveRTTokenizer"},
+                {"name": "ConveRTFeaturizer"},
+                {"name": "EmbeddingIntentClassifier", EPOCHS: 3},
```
Comment (dakshvar22) [inline tests/nlu/conftest.py]: Random seed missing

File: tests/nlu/extractors/test_crf_entity_extractor.py [hunk 4]
```diff
@@ -41,7 +42,16 @@ def test_crf_extractor(spacy_nlp, ner_crf_pos_feature_config):
...
 
-    extractor = CRFEntityExtractor(component_config=ner_crf_pos_feature_config)
+    extractor = CRFEntityExtractor(
+        component_config={
+            "features": [
+                ["low", "title", "upper", "pos", "pos2"],
+                ["low", "suffix3", "suffix2", "upper", "title", "digit", "pos", "pos2"],
+                ["low", "title", "upper", "pos", "pos2"],
+            ],
+            RANDOM_SEED: 1,
```
Comment (dakshvar22) [inline tests/nlu/extractors/test_crf_entity_extractor.py]: Do we need to add smaller number of epochs here? i remember this worked with 75 epochs the last time I tested with a particular speed. Can leave it out if the speed difference is marginal.

File: tests/nlu/extractors/test_entity_synonyms.py [hunk 5]
```diff
@@ -2,10 +2,17 @@
 from tests.nlu import utilities
```
Comment (dakshvar22) [inline tests/nlu/extractors/test_entity_synonyms.py]: There is a file `tests/nlu/extractors/test_synonyms.py` with a test named `test_entity_synonyms` and there also exists a file named `test_entity_synonyms.py` . Can we consolidate them?

Comment (dakshvar22) [inline tests/nlu/extractors/test_entity_synonyms.py]: isn't used
```suggestion
```

File: tests/nlu/test_evaluation.py [hunk 6]
```diff
@@ -315,7 +302,7 @@ def test_run_cv_evaluation():
     assert len(entity_results.test["CRFEntityExtractor"]["F1-score"]) == n_folds
 
 
-def test_run_cv_evaluation_with_response_selector():
+def test_run_cv_evaluation_with_response_selector(supervised_embeddings_config):
```
Comment (dakshvar22) [inline tests/nlu/test_evaluation.py]: supervised_embeddings_config doesn't have response selector component in it

Comment (tabergma) [reply tests/nlu/test_evaluation.py]: the config it not used, will remove it
